storage/remote/client.go

```diff
@@ -272,6 +284,7 @@ func (c *Client) Store(ctx context.Context, req []byte, attempt int) (WriteRespo
 	httpReq.Header.Add("Content-Encoding", string(c.writeCompression))
 	httpReq.Header.Set("Content-Type", remoteWriteContentTypeHeaders[c.writeProtoMsg])
 	httpReq.Header.Set("User-Agent", UserAgent)
+	httpReq.Header.Set("X-Prometheus-User-Agent", internalUserAgent)
```

Comment:
Calling it something like "library version" might be more descriptive?
Also, would the internal Go version be even more useful?

Reply:
Updated to be `X-Prometheus-Library-Version`.

The internal Go version might also be useful but the goal of this PR was to ensure that the current/default value was available somewhere in the request if the client decides to overwrite the modifiable `UserAgent` variable. We can always add the internal Go version as an extra header if we think it is necessary.